src/boundaries.rs

```diff
@@ -381,3 +431,164 @@ fn test_build_two_boundary_closed() {
         assert!(false); //this should not happen
     }
 }
+
+#[test]
+fn test_build_one_donut_boundary() {
+    use geo::algorithm::area::Area;
+    let mut builder = osm_builder::OsmBuilder::new();
+    let rel_id = builder
+        .relation()
+        .outer(vec![
+            named_node(0.0, 0.0, "start"),
+            named_node(4.0, 0.0, "1"),
+            named_node(4.0, 4.0, "2"),
+            named_node(0.0, 4.0, "3"),
+            named_node(0.0, 0.0, "start"),
+        ])
+        .inner(vec![
+            named_node(1.0, 1.0, "other_start"),
+            named_node(2.0, 1.0, "11"),
+            named_node(2.0, 2.0, "12"),
+            named_node(1.0, 2.0, "13"),
+            named_node(1.0, 1.0, "other_start"),
+        ])
+        .relation_id
+        .into();
+    if let osmpbfreader::OsmObj::Relation(ref relation) = builder.objects[&rel_id] {
+        let multipolygon = build_boundary(&relation, &builder.objects);
+        assert!(multipolygon.is_some());
+        let multipolygon = multipolygon.unwrap();
+        assert_eq!(multipolygon.0.len(), 1);
+        assert_eq!(multipolygon.area(), 15.);
+    } else {
+        assert!(false); //this should not happen
+    }
+}
+
+#[test]
+fn test_build_two_boundaries_with_one_hole() {
+    use geo::algorithm::area::Area;
+    let mut builder = osm_builder::OsmBuilder::new();
+    let rel_id = builder
+        .relation()
+        .outer(vec![
+            named_node(0.0, 0.0, "start"),
+            named_node(4.0, 0.0, "1"),
+            named_node(4.0, 4.0, "2"),
+            named_node(0.0, 4.0, "3"),
+            named_node(0.0, 0.0, "start"),
+        ])
+        .inner(vec![
+            named_node(1.0, 1.0, "other_start"),
+            named_node(2.0, 1.0, "11"),
+            named_node(2.0, 2.0, "12"),
+            named_node(1.0, 2.0, "13"),
+            named_node(1.0, 1.0, "other_start"),
+        ])
+        .outer(vec![
+            named_node(4.0, 4.0, "yet_another_start"),
+            named_node(8.0, 4.0, "4"),
+            named_node(8.0, 8.0, "5"),
+            named_node(4.0, 8.0, "6"),
+            named_node(0.0, 0.0, "yet_another_start"),
```

Comment:
It should be `4.0, 4.0`, I suppose.

Reply:
I can change it if you want, but it works like this, because it is a named_node that is already defined.